src/hooks/useThemeProducts.ts

```diff
@@ -23,11 +24,20 @@ export function useThemeProducts(themeId: number | undefined) {
     hasNextPage,
     fetchNextPage,
     isFetchingNextPage,
-  } = useGetThemeProducts(themeId || 0) as any;
+  } = useGetThemeProducts(themeId || 0) as {
+    data: { pages: ThemeProductsResponse[] } | undefined;
+    isLoading: boolean;
+    hasNextPage: boolean | undefined;
+    fetchNextPage: () => void;
+    isFetchingNextPage: boolean;
+  };
 
   useEffect(() => {
-    if (themeError && (themeError as any)?.response?.status === 404) {
-      navigate(RouterPath.HOME, { replace: true });
+    if (themeError) {
+      const apiError = themeError as ApiError;
+      if (apiError?.response?.status === 404) {
```

Comment:
axios에 응답 코드가 정의되어 있는데 이걸 사용해봐도 좋을 것 같아요!
```suggestion
      if (apiError?.response?.status === HttpStatusCode.NotFound) {
```